Accessing the API as an integration

naderman opened the pull request:
- Add JWT authentication introduced with GitHub integrations
- Add endpoint for Installation access tokens required for API use from GitHub integrations

naderman commented:
Currently I'm using the [locobucci/jwt](https://github.com/lcobucci/jwt/tree/3.2.0) library to create a JSON Web Token. Not sure if I should add this as a dependeny here and provide a convenience method, or if this should just go into the documentation? To actually get an access token for use with the API the following is currently necessary:

```php
        $jwt = (new Builder)
            ->setIssuer($integrationId)
            ->setIssuedAt(time())
            ->setExpiration(time() + 60)
            ->sign(new Sha256(),  (new Keychain)->getPrivateKey($pemPrivateKeyPath))
            ->getToken();

        $github->authenticate($jwt, null, Github\Client::AUTH_JWT);

        $token = $github->api('installations')->createAccessToken($installationId);
        $github->authenticate($token['token'], null, Github\Client::AUTH_HTTP_TOKEN);
```

naderman commented:
Not sure what the build failures are about.

naderman commented:
@cursedcoder what are your thoughts on my question above? Otherwise this works as-is, so would be great if you could merge it. There's probably more features one could add later, but this is enough to work with github integrations.

cursedcoder commented:
@naderman could you please add a doc with this content https://github.com/KnpLabs/php-github-api/pull/439#issuecomment-249991521?

cursedcoder commented on lib/Github/Api/Installations.php:
```diff
@@ -0,0 +1,28 @@
+        if ($userId) {
+            $paramters['user_id'] = $userId;
+        }
+        return $this->post('/installations/'.rawurlencode($installationId).'/access_tokens', $parameters);
```
should be separated with 1 space line above

cursedcoder replied:
https://styleci.io/analyses/8mryaW there's a proper diff how it should look lile

naderman replied:
Haha sorry, thought you meant you wanted spaces around concetanation, will add a blank line ;-) Will comment when I finished writing some docs for this.

naderman commented:
@cursedcoder Alright I added some documentation, and renamed the endpoint to Integrations as per the current API docs. There will likely be a sub object installations at some point. I tested this against the actual API and works for me.

cursedcoder commented:
thanks !
